xarray/core/merge.py

```diff
@@ -365,6 +365,19 @@ def merge_data_and_coords(data, coords, compat='broadcast_equals',
     return merge_core(objs, compat, join, explicit_coords=explicit_coords)
 
 
+def assert_valid_explicit_coords(variables, dims, explicit_coords):
+    """Validate explicit coordinate names/dims.
+
+    Raise a MergeError if an explicit coord shares a name with a dimension
+    but is comprised of arbitrary dimensions.
+    """
+    for coord_name in explicit_coords:
+        if coord_name in dims and variables[coord_name].dims != (coord_name,):
+            raise MergeError(
+                'coordinate %s shares a name with a dimension but '
+                'includes at least one arbitrary dimensions' % coord_name)
```

Comment:
"includes at least one arbitrary dimensions" is a little confusing to me.

Let's try to be totally explicit here, even if the error message needs to be longer. Maybe:

> coordinate X shares a name with a dataset dimension, but is not a 1D variable along that dimension. This is disallowed by the xarray data model.

Reply:
done.